### PR TITLE
Comments

### LINKED ISSUE #22475: Clarify Layout-thread exit workflow
It appears to me that the `ExitNow` message, handled [here](https://github.com/servo/servo/blob/6731ddd29f40f05ebb65fef0b1164c8632a7872a/components/layout_thread/lib.rs#L814), is in fact never received at that stage. Rather, it seems that it is only received and handled inside `prepare_to_exit`. 

If that is indeed the case, we should probably add an expectation that it is never received at the initial stage(where only a `PrepareToExit` message should be received). This could make the workflow more explicit. 

The actionable item here would be doing some investigation by adding a `panic`, running the test suite(on the server from a PR), and seeing if the message is received at all. If it is not, we can leave the panic(or maybe turn it into a `warn`) as a form of documentation and to ensure the message is not sent due to a code change later, unless the workflow is meant to be changed.

### PR DESCRIPTION
**Pull request by aditj**
---
Added comments to clarify layout-thread exit workflow (Specifically when will a Exit message be received )

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #22475 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because only comments are added.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22623)

### CODE REVIEW

**Comment by KiChjang** [comment]
This PR contains commits that are from #22431. Could you please remove them?

**Comment by aditj** [comment]
Can you help me do that?

File: components/layout_thread/lib.rs [hunk 1]
```diff
@@ -789,6 +789,7 @@ impl LayoutThread {
                 self.prepare_to_exit(response_chan);
                 return false;
             },
+            //Recieving the Exit message at this stage only happens when layout is undergoing a "force exit".
```
Comment (gterzian) [inline components/layout_thread/lib.rs]: nit: Typo and lack of leading space, should be `// Receiving`

**Comment by gterzian** [comment]
`git rebase -i HEAD~3`, and comment out the commits that you want to remove by adding a `#` in front of them. That is assuming you want to remove those commits from this branch.

**Pull request closed without merging**